File: client/src/main/scala/ada/client/output/DefaultClientOutput.java
package ada.client.output;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Value;

import java.io.PrintStream;

/**
 * @author Michael Wellner (michael.wellner@de.ibm.com)
 */
@Value
@EqualsAndHashCode
@AllArgsConstructor(staticName = "apply")
public class DefaultClientOutput implements ClientOutput {

    private final PrintStream out;

    public static DefaultClientOutput apply() {
        return apply(System.out);
    }

    @Override
    public PrintStream getPrintStream() {
        return out;
    }

    @Override
    public void println(String message, Object... args) {
        out.println(String.format(message, args));
    }

}

File: client/src/main/scala/ada/client/output/DefaultClientOutput.scala
package ada.client.output

import java.io.PrintStream

final case class DefaultClientOutput(ps: PrintStream) extends ClientOutput {

  override def getPrintStream: PrintStream = ps

  override def println(message: String, args: Object*): Unit = {
    val s = String.format(message, args)
    println(s)
  }

  override def println(message: String): Unit = System.out.println(message)

  override def print(message: String, args: Object*): Unit = print(String.format(message, args))

  override def print(message: String): Unit = System.out.print(message)

}

object DefaultClientOutput {

  def apply(): DefaultClientOutput = DefaultClientOutput(System.out)

}
